core/lib/l1_contract_interface/src/i_executor/structures/commit_batch_info.rs

```diff
@@ -1,32 +1,35 @@
+use std::sync::Arc;
+
+use zkevm_test_harness_1_4_2::kzg::KzgSettings;
 use zksync_types::{
     commitment::{pre_boojum_serialize_commitments, serialize_commitments, L1BatchWithMetadata},
     ethabi::Token,
+    pubdata_da::PubdataDA,
     web3::{contract::Error as Web3ContractError, error::Error as Web3ApiError},
     U256,
 };
 
-use crate::Tokenizable;
+use crate::{
+    i_executor::commit::kzg::{KzgInfo, ZK_SYNC_BYTES_PER_BLOB},
+    Tokenizable,
+};
+
+/// These are used by the L1 Contracts to indicate what DA layer is used for pubdata
+const PUBDATA_SOURCE_CALLDATA: u8 = 0;
+const PUBDATA_SOURCE_BLOBS: u8 = 1;
 
 /// Encoding for `CommitBatchInfo` from `IExecutor.sol`
 #[derive(Debug)]
-pub struct CommitBatchInfo<'a>(pub &'a L1BatchWithMetadata);
-
-impl<'a> Tokenizable for CommitBatchInfo<'a> {
-    fn from_token(_token: Token) -> Result<Self, Web3ContractError>
-    where
-        Self: Sized,
-    {
-        // Currently there is no need to decode this struct.
-        // We still want to implement `Tokenizable` trait for it, so that *once* it's needed
-        // the implementation is provided here and not in some other inconsistent way.
-        Err(Web3ContractError::Api(Web3ApiError::Decoder(
-            "Not implemented".to_string(),
-        )))
-    }
+pub struct CommitBatchInfo<'a>(
+    pub &'a L1BatchWithMetadata,
```

Comment:
nit: I'd use named fiels, so that we can say things like self.kzg instead of self.2